File: tests/sdk/nodejs/test/secrets.test.mjs
import { test } from "node:test";
import assert from "node:assert/strict";
import {
  CreateSecretCommand,
  DeleteSecretCommand,
  GetSecretValueCommand,
} from "@aws-sdk/client-secrets-manager";
import { newSecrets, requireReady, uniquePrefix } from "../lib/helpers.mjs";

test("Secrets Manager round-trip", async (t) => {
  if (!(await requireReady(t))) return;
  const secrets = newSecrets();
  const prefix = uniquePrefix();
  const name = `lab/${prefix}/api-key`;
  const secret = JSON.stringify({ apiKey: `key-${prefix}` });

  await secrets.send(
    new CreateSecretCommand({
      Name: name,
      SecretString: secret,
    }),
  );
  t.after(async () => {
    try {
      await secrets.send(
        new DeleteSecretCommand({
          SecretId: name,
          ForceDeleteWithoutRecovery: true,
        }),
      );
    } catch {
      /* ignore */
    }
  });

  const got = await secrets.send(
    new GetSecretValueCommand({ SecretId: name }),
  );
  assert.equal(got.SecretString, secret);

  await secrets.send(
    new DeleteSecretCommand({
      SecretId: name,
      ForceDeleteWithoutRecovery: true,
    }),
  );
});

File: tests/sdk/go/secrets_test.go
package sdk_test

import (
	"context"
	"testing"

	"github.com/aws/aws-sdk-go-v2/aws"
	"github.com/aws/aws-sdk-go-v2/service/secretsmanager"
)

func TestSecretsManagerRoundTrip(t *testing.T) {
	requireReady(t)
	cfg := loadAWSConfig(t)
	client := newSecrets(t, cfg)
	ctx := context.Background()
	prefix := uniquePrefix(t)
	name := "lab/" + prefix + "/api-key"
	secret := `{"apiKey":"key-` + prefix + `"}`

	_, err := client.CreateSecret(ctx, &secretsmanager.CreateSecretInput{
		Name:         aws.String(name),
		SecretString: aws.String(secret),
	})
	if err != nil {
		t.Fatalf("CreateSecret: %v", err)
	}
	t.Cleanup(func() {
		_, _ = client.DeleteSecret(ctx, &secretsmanager.DeleteSecretInput{
			SecretId:                   aws.String(name),
			ForceDeleteWithoutRecovery: aws.Bool(true),
		})
	})

	got, err := client.GetSecretValue(ctx, &secretsmanager.GetSecretValueInput{
		SecretId: aws.String(name),
	})
	if err != nil {
		t.Fatalf("GetSecretValue: %v", err)
	}
	if aws.ToString(got.SecretString) != secret {
		t.Fatalf("GetSecretValue=%q want=%q", aws.ToString(got.SecretString), secret)
	}

	_, err = client.DeleteSecret(ctx, &secretsmanager.DeleteSecretInput{
		SecretId:                   aws.String(name),
		ForceDeleteWithoutRecovery: aws.Bool(true),
	})
	if err != nil {
		t.Fatalf("DeleteSecret: %v", err)
	}
}
